### ISSUE TITLE
Broken Test: Metricbeat Generator

### ISSUE THREAD

**Issue by urso** [elastic/beats#18132]
## Flaky Test

* **Test Name:** Metricbeat Generator test
* **Branch:** master
* **Notes:** the test tries to execute a mage target that does not exist anymore.

Test is run via `make -C generator/_template/metricbeat test`

### Stack Trace

```
[2020-04-29T18:54:32.807Z] =======================

[2020-04-29T18:54:32.807Z] Your custom beat is now available as /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat

[2020-04-29T18:54:32.807Z] =======================

[2020-04-29T18:54:32.807Z] cd /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat ; \

[2020-04-29T18:54:32.807Z] export PATH=${GOPATH}/bin:${PATH}; \

[2020-04-29T18:54:32.807Z] git config user.email "beats-jenkins@test.com" || exit 1 ; \

[2020-04-29T18:54:32.807Z] git config user.name "beats-jenkins" || exit 1 ; \

[2020-04-29T18:54:32.807Z] make check CHECK_HEADERS_DISABLED=y || exit 1 ; \

[2020-04-29T18:54:32.807Z] make || exit 1 ; \

[2020-04-29T18:54:32.807Z] make unit

[2020-04-29T18:54:32.807Z] make[1]: Entering directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:54:32.807Z] mage check

[2020-04-29T18:54:35.359Z] >> fmt - goimports: Formatting Go code

[2020-04-29T18:54:36.308Z] >> check: Checking source code for common problems

[2020-04-29T18:55:32.631Z] make[1]: Leaving directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:55:32.631Z] make[1]: Entering directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:55:32.631Z] Use mage rather than make. Here are the available mage targets:

[2020-04-29T18:55:32.631Z] Targets:

[2020-04-29T18:55:32.631Z]   build               builds the Beat binary.

[2020-04-29T18:55:32.632Z]   buildGoDaemon       builds the go-daemon binary (use crossBuildGoDaemon).

[2020-04-29T18:55:32.632Z]   check               formats code, updates generated content, check for common errors, and checks for any modified files.

[2020-04-29T18:55:32.632Z]   clean               cleans all generated files and build artifacts.

[2020-04-29T18:55:32.632Z]   collectAll          generates the docs and the fields.

[2020-04-29T18:55:32.632Z]   config              generates both the short/reference/docker configs.

[2020-04-29T18:55:32.632Z]   createMetricset     creates a new metricset.

[2020-04-29T18:55:32.632Z]   crossBuild          cross-builds the beat for all target platforms.

[2020-04-29T18:55:32.632Z]   dashboards          collects all the dashboards and generates index patterns.

[2020-04-29T18:55:32.632Z]   fields              generates a fields.yml for the Beat.

[2020-04-29T18:55:32.632Z]   fieldsDocs          generates docs/fields.asciidoc containing all fields (including x-pack).

[2020-04-29T18:55:32.632Z]   fmt                 formats source code (.go and .py) and adds license headers.

[2020-04-29T18:55:32.632Z]   golangCrossBuild    build the Beat binary inside of the golang-builder.

[2020-04-29T18:55:32.632Z]   imports             generates an include/list.go file containing a import statement for each module and dataset.

[2020-04-29T18:55:32.632Z]   package             packages the Beat for distribution.

[2020-04-29T18:55:32.632Z]   test                runs all available tests

[2020-04-29T18:55:32.632Z]   update              is an alias for running fields, dashboards, config.

[2020-04-29T18:55:32.632Z]   vendorUpdate        updates elastic/beats/v7 in the vendor dir

[2020-04-29T18:55:32.632Z] make[1]: Leaving directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:55:32.632Z] make[1]: Entering directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:55:32.632Z] make[1]: *** No rule to make target 'unit'.  Stop.

[2020-04-29T18:55:32.632Z] make[1]: Leaving directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-16784/src/beatpath/testmetricbeat'

[2020-04-29T18:55:32.632Z] ../../common/Makefile:13: recipe for target 'test' failed

```

**Comment by blakerouse** [comment]
Yes I have been trying to fix it, working on it here.

https://github.com/elastic/beats/pull/18086

**Comment by blakerouse** [comment]
run beats-ci/package

**Comment by blakerouse** [comment]
Fixed by #18086.
